elixir/apps/web/lib/web/live/resources/edit.ex

```diff
@@ -4,14 +4,15 @@ defmodule Web.Resources.Edit do
   alias Domain.Gateways
   alias Domain.Resources
 
-  def mount(%{"id" => id}, _session, socket) do
+  def mount(%{"id" => id} = params, _session, socket) do
     with {:ok, resource} <-
            Resources.fetch_resource_by_id(id, socket.assigns.subject, preload: :gateway_groups),
          {:ok, gateway_groups} <- Gateways.list_groups(socket.assigns.subject) do
       form = Resources.change_resource(resource, socket.assigns.subject) |> to_form()
 
-      {:ok, assign(socket, resource: resource, gateway_groups: gateway_groups, form: form),
+      {:ok, assign(socket, resource: resource, form: form, params: Map.take(params, ["site_id"])),
        temporary_assigns: [
+         gateway_groups: gateway_groups,
```

Comment:
I think I may need some help understanding this.  

I was always under the impression that the `temporary_assigns` list contained the default values for the given keys and those variables would be set back to the default value after each update, so as not to consume more memory on the process.  So if you had a list or map that you didn't want to keep around in memory you could set it back to an empty list or empty map, like the example in the documentation:

```
def mount(_params, _session, socket) do
  socket = assign(socket, :messages, load_last_20_messages())
  {:ok, socket, temporary_assigns: [messages: []]}
end
```

Is there some type of default behavior that happens when you assign to `temporary_assigns` in the way you have it here?  If there's not, wouldn't this just store all the gateway groups as the default and take up just as much memory?

Reply:
@bmanifold you are right, I completely misread the documentation and assumed `temporary_assigns` are a single-render assigns that are discarded afterwards. What I really wanted here is to assign, set temporary assign to an empty value and add `phx-update=ignore` to an element so that we don't need to keep large state in memory.